### PR TITLE
Named Data Networking (NDN) Tools

### PR DESCRIPTION
**Pull request by MostAwesomeDude**
###### Motivation for this change

I want to play with NDN.

###### Things done

- [ ] Tested using sandboxing
  ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS,
    or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file)
    on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] Linux
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

I had to use versions from git because this stuff is so bleeding-edge.

### CODE REVIEW

**Comment by MostAwesomeDude** [comment]
Hm. These actually do have releases; I'm going to fix that and rebase.

File: pkgs/development/libraries/ndn-cxx/default.nix [hunk 1]
```diff
@@ -1,30 +1,36 @@
...
-      --with-examples \
       --prefix=$out
   '';
   buildPhase = ''
     ./waf
   '';
+  # To do battle with the tests, add --with-tests to the configure above. ~ C.
+  doCheck = false;
+  checkPhase = ''
+    LDD_LIBRARY_PATH=build/ build/unit-tests
```
Comment (0xABAB) [inline pkgs/development/libraries/ndn-cxx/default.nix]: Don't you mean LD_LIBRARY_PATH?

File: pkgs/servers/named-data/nfd/default.nix [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+    url = "https://github.com/named-data/NFD.git";
+    rev = "8eb5f8f3ed6c3f55d4c334869fee73d3ec762eec";
+    sha256 = "0k6lz7p8vpf97k3nqy8s6fyjg6iqvgbx8jwhrdwpfsf1kwbiw291";
+  };
+  buildInputs = [ libpcap doxygen boost pkgconfig python pythonPackages.sphinx
+                  openssl ndn-cxx ];
+  preConfigure = ''
+    patchShebangs waf
+    ./waf configure \
+      --boost-includes=${boost.dev}/include \
```
Comment (0xABAB) [inline pkgs/servers/named-data/nfd/default.nix]: Needs quoting.

**Review by 0xABAB — CHANGES_REQUESTED** [review]
Minor stuff. Thanks for your contribution.

File: pkgs/development/libraries/ndn-cxx/default.nix (hunk 1 above)
Comment (MostAwesomeDude) [inline pkgs/development/libraries/ndn-cxx/default.nix]: Yes. Fixed. The tests still don't pass, of course.

File: pkgs/servers/named-data/nfd/default.nix (hunk 2 above)
Comment (MostAwesomeDude) [inline pkgs/servers/named-data/nfd/default.nix]: Just this one, or in all of the configures?

Comment (0xABAB) [inline pkgs/servers/named-data/nfd/default.nix]: All locations.

File: pkgs/tools/networking/named-data/repo-ng/default.nix [hunk 3]
```diff
@@ -0,0 +1,42 @@
+{ stdenv, fetchgit, openssl, doxygen, boost, sqlite, pkgconfig,
+  python, pythonPackages, libpcap, ndn-cxx }:
+let
+  version = "2b7b83";
+in
+stdenv.mkDerivation {
+  name = "ndn-tools-0.1-${version}";
```
Comment (Mic92) [inline pkgs/tools/networking/named-data/repo-ng/default.nix]: name is probably wrong.

Comment (MostAwesomeDude) [inline pkgs/tools/networking/named-data/repo-ng/default.nix]: What a mistake! Thanks.

**Comment by MostAwesomeDude** [comment]
I've addressed all feedback and am ready for merge. `nfd` works, at least. I haven't gotten far enough to test the "ndnsec" toolset. `repo-ng` is cantankerous, which is unsurprising since it is still immature.

**Review by 0xABAB — APPROVED** [review]

**Review by matthewbauer — CHANGES_REQUESTED** [review]
Fix merge conflicts

**Comment by mmahut** [comment]
Are there any updates on this pull request, please?

**Comment by MostAwesomeDude** [comment]
Hi, I rebased the branch and also bumped the versions of all new packages. It seems that, at some point, a derivation based on my ndn-cxx work got merged, so now the branch's history is a bit messy. I can redo all of the commits, if desired. I haven't been poking at this stuff for a while, though.

**Comment by sjmackenzie** [comment]
@matthewbauer I believe the merge conflicts were resolved. Can you merge please?

File: pkgs/servers/named-data/nfd/default.nix [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+  name = "nfd-${version}";
+  src = fetchFromGitHub {
+    owner = "named-data";
+    repo = "NFD";
+    rev = "NFD-${version}";
+    sha256 = "0y86x9h6wk2n9cfxgnmm0h89kjdv01i7pbf189354yc2ylndmbgh";
+  };
+  buildInputs = [ wafHook
+                  libpcap doxygen boost pkgconfig python pythonPackages.sphinx
+                  openssl ndn-cxx ];
```
Comment (Mic92) [inline pkgs/servers/named-data/nfd/default.nix]:
```
nativeBuildInputs = [ wafHook doxygen pkgconfig python pythonPackages.sphinx ];
```

File: pkgs/tools/networking/named-data/ndn-tools/default.nix [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+  name = "ndn-tools-${version}";
+
+  src = fetchFromGitHub {
+    owner = "named-data";
+    repo = "ndn-tools";
+    rev = "ndn-tools-${version}";
+    sha256 = "1sjhyhc9ww38vyf3qmjzh5c4c0jc43ypzjmpqwsbq8y5zzln0gfx";
+  };
+
+  buildInputs = [ wafHook libpcap openssl boost pkgconfig python pythonPackages.sphinx git ndn-cxx ];
```
Comment (Mic92) [inline pkgs/tools/networking/named-data/ndn-tools/default.nix]:
```
nativeBuildInputs = [ wafHook pkgconfig python pythonPackages.sphinx git ];
```

btw. why is `git` required here?

File: pkgs/tools/networking/named-data/repo-ng/default.nix [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+stdenv.mkDerivation {
+  name = "repo-ng-0.1-${version}";
+  src = fetchFromGitHub {
+    owner = "named-data";
+    repo = "repo-ng";
+    rev = "3c843162f812ea22972d82fe7a834fac3f3ae2d5";
+    sha256 = "0pdinvq3cf458bmm2mxdlidrmmr38in1563ra8q0ch3wp45va2di";
+  };
+  buildInputs = [ wafHook libpcap openssl doxygen boost sqlite pkgconfig
+                  python pythonPackages.sphinx ndn-cxx ];
```
Comment (Mic92) [inline pkgs/tools/networking/named-data/repo-ng/default.nix]:
```
nativeBuildInputs = [ wafHook doxygen pkgconfig python pythonPackages.sphinx ];
```

File: pkgs/development/libraries/ndn-cxx/default.nix [hunk 7]
```diff
@@ -1,24 +1,37 @@
...
     owner = "named-data";
     repo = "ndn-cxx";
-    rev = "a3bf4319ed483a4a6fe2c96b79ec4491d7217f00";
-    sha256 = "076jhrjigisqz5n8dgxwd5fhimg69zhm834m7w9yvf9afgzrr50h";
+    rev = "ndn-cxx-${version}";
+    sha256 = "06q4bsl63bn5h9csdyz883qpak67511w5d7xl3z5364pmjfwvji2";
   };
+
   nativeBuildInputs = [ pkgconfig wafHook ];
   buildInputs = [ openssl doxygen boost sqlite python pythonPackages.sphinx];
```
Comment (Mic92) [inline pkgs/development/libraries/ndn-cxx/default.nix]: also doxygen, python and sphinx should be in nativeBuildInputs

**Comment by roberth** [comment]
ndn-cxx has been updated to 0.7.1 in master. You could rebase `-i` and omit the update, then add yourself to the maintainers in a new commit.

**Pull request closed without merging**